llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -739,6 +766,29 @@ bool AIEPostRASchedStrategy::tryCandidate(SchedCandidate &Cand,
       return TryCand.Reason != NoCand;
     }
 
+    // Special heuristics for loops.
+    // Note that they aren't used for the first fixpoint iteration: this is
+    // currently a workaround because we want a very optimistic schedule in that
+    // first iteration. That is because it decides the slot assignments for
+    // multi-slot instructions. This rule can probably be deleted once the
+    // loop-aware scheduler knows how to reassign those.
+    const BlockState &BS = getInterBlock().getBlockState(CurMBB);
+    if (UseLoopHeuristics && BS.Kind == AIE::BlockType::Loop &&
+        BS.getRegions().size() == 1 && BS.FixPoint.NumIters > 0) {
```

Comment:
Yeah right. Note that getRegions().size() is unstable as long as BS.FixPoint.NumIters == 0